salt/modules/salty_swarm.py

```diff
@@ -0,0 +1,269 @@
+# -*- coding: utf-8 -*-
+"""
+Docker Swarm Module using Docker's Python SDK
+=============================================
+
+:codeauthor: Tyler Jones <jonestyler806@gmail.com>
+
+.. versionadded:: 2017.7.2
+
+The Docker Swarm Module is used to manage and create Docker Swarms.
+
+Dependencies
+============
+
+- Docker installed on the host
+- Docker python sdk >= 2.5.1
+
+Docker Python SDK
+=================
+pip install -U docker
+
+More information: https://docker-py.readthedocs.io/en/stable/
+"""
+from __future__ import absolute_import
+import docker
+import salt.config
+import salt.loader
+import json
+__opts__ = salt.config.minion_config('/etc/salt/minion')
+__grains__ = salt.loader.grains(__opts__)
+client = docker.from_env()
+server_name = __grains__['id']
```

Comment:
You should not be specifying `__opts__` or `__grains__`, these will be injected by the loader.

You can specify the client and server_name in the `__context__` in  the `__init__` function.

Reply:
@gtmanfred could you elaborate more or show a example?

Reply:
you should be able to replace all this with.

```
def __init__():
    __context__['client'] = docker.from_env()
    __context__['server_name'] = __grains__['id']
```

But you should probably look at what the `salt.modules.dockermod` module is doing, and replicate how it is setting up the docker stuff, so that it doesn't have to be environment variables but is set in /etc/salt/minion or in grains or in pillars.

https://github.com/saltstack/salt/blob/2017.7/salt/modules/dockermod.py#L321-L366

Reply:
@gtmanfred Thanks! I will take a look.

Reply:
@gtmanfred  I added the `__init__()` but i am working on looking at the dockermod module to see what I can do with it.